keys-core/src/main/java/io/kidsfirst/keys/core/LambdaRequestHandler.java

```diff
@@ -0,0 +1,90 @@
+package io.kidsfirst.keys.core;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import io.jsonwebtoken.Jwt;
+import io.kidsfirst.keys.core.model.LambdaResponse;
+import io.kidsfirst.keys.core.utils.JWTUtils;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+public abstract class LambdaRequestHandler implements RequestStreamHandler {
+
+  private static JSONParser parser = new JSONParser();
+
+  public abstract String processEvent(JSONObject event, String userId) throws IllegalArgumentException, ParseException;
+
+  public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
+    LambdaResponse resp = new LambdaResponse();
+    String data = "";
+
+
+    try(BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        OutputStreamWriter writer = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8)) {
+
+      JSONObject event = (JSONObject)parser.parse(reader);
+
+      String auth = (String)((JSONObject)event.get("headers")).getOrDefault("Authorization","");
+
+      String userId = null;
+      try {
```

Comment:
Its probably a good idea to put the whole try-catch block for getting userId into its own static method. 
That way you can just call
```java
String userId = getUserId(event) 
```
and the reader knows exactly what is going on without having to have full knowledge of the exception handling.